### DISCUSSION TITLE
PropertyPlot Not Solving for States Correctly

### DISCUSSION THREAD

**Discussion by siriusblakk** [CoolProp/CoolProp#2321] [Q&A]
My property plots for R120 and R744 are erroring out. The plots on this [page](http://www.coolprop.org/coolprop/python-plotting.html) looks complete while mine looks like: 
![image](https://github.com/CoolProp/CoolProp/assets/148289450/143cd18c-6398-4990-b9e3-765df146b0e4)

Code:
![image](https://github.com/CoolProp/CoolProp/assets/148289450/1b2941c1-b3c1-4390-a607-2d756e6e98dd)

Ideas?

**Comment by fwitte** [comment]
Some time ago I created a small package for seamless integration of fluid property diagrams with matplotlib interface (and being able to label the isolines and some other reasons) using CoolProp: https://fluprodia.readthedocs.io/en/latest/usage.html. The API let's you select which isolines to calculate and once calculated you can create logp,h T,s h,s... diagrams. All data are availble through the `diagram.entropy`, `diagram.enthalpy`, `diagram.quality`, ... dictionaries which hold all the other properties for that isoline, e.g. `diagram.entropy[5000.0]['h']` to get all enthalpy values corresponding to the `5000.0 J/kgK` isentropic from the example below.

For example for water:

![Ts_diagram_H2O](https://github.com/CoolProp/CoolProp/assets/20252362/f8e597bf-0637-4100-89f3-a1f5809cb3a3)
![logph_diagram_H2O](https://github.com/CoolProp/CoolProp/assets/20252362/c21d75cd-42ff-422b-b375-33e1045aa735)

```python
from fluprodia import FluidPropertyDiagram
import matplotlib.pyplot as plt
import numpy as np

diagram = FluidPropertyDiagram(fluid='H2O')
diagram.set_unit_system(T='°C', h='kJ/kg', p='bar')
Q = np.linspace(0, 1, 11)
T = np.arange(25, 501, 25)
p = np.geomspace(0.01, 1000, 6) * 1e5
v = np.geomspace(0.001, 10, 5)
s = np.linspace(1000, 10000, 10)
h = np.linspace(0, 3600, 19)
diagram.set_isolines(Q=Q, T=T, p=p, v=v, s=s, h=h)
diagram.calc_isolines()

fig, ax = plt.subplots(1, figsize=(8, 5))
diagram.draw_isolines(diagram_type='logph', fig=fig, ax=ax, x_min=0, x_max=3000, y_min=0.01, y_max=1000)
plt.tight_layout()
fig.savefig('logph_diagram_H2O.png', dpi=300)

fig, ax = plt.subplots(1, figsize=(8, 5))
diagram.draw_isolines(diagram_type='Ts', fig=fig, ax=ax, x_min=0, x_max=8000, y_min=0, y_max=700)
plt.tight_layout()
fig.savefig('Ts_diagram_H2O.png', dpi=300)
```

If somebody is interested in investing some time and if it is desired, this may also be integrated into CoolProp?!

Best
